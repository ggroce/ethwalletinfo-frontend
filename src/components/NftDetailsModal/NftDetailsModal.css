.nftdetails__background--darken {
  background-color: rgba(20, 20, 20, 0.7);
  width: 100vw;
  height: 100vh;
  z-index: 0;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  position: fixed;
  overflow: auto;
}

.nftdetails__wrapper--center {
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.nftdetails__outerwrapper {
  width: 75vw;
  max-height: 90vh;
  z-index: 10;
  border: 2px solid;
  border-radius: 10px;
  border-color: rgb(151, 141, 141);
  background-color: #191b20;
  overflow: hidden;
  overflow-y: auto;
  scrollbar-width: thin;
}

.nftdetails__innerwrapper {
  margin-left: auto;
  margin-right: auto;
  width: 90%;
}

.nftdetails__img {
  display: block;
  margin-left: auto;
  margin-right: auto;
  max-width: 18rem;
  max-height: 18rem;
  width: auto;
  height: auto;
}

.nftdetails__outerwrapper::-webkit-scrollbar {
  width: 7px;
}

.nftdetails__outerwrapper::-webkit-scrollbar-track {
  -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.3);
}

.nftdetails__outerwrapper::-webkit-scrollbar-thumb {
  background-color: darkgrey;
  outline: 1px solid slategrey;
}
