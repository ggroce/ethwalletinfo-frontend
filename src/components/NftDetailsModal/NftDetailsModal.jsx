import './NftDetailsModal.css';

const NftDetailsModal = ({ setShowDetails, props }) => {
  return (
    <>
      <div
        className="nftdetails__background--darken"
        onClick={() => setShowDetails(false)}
      >
        <div className="nftdetails__wrapper--center">
          <div className="nftdetails__outerwrapper">
            <div className="nftdetails__innerwrapper">
              <h5>{props?.meta.name}</h5>
              <img
                className="nftdetails__img"
                src={props?.meta.image.url.ORIGINAL}
                loading="lazy"
                title="Picture of NFT"
                alt={props?.meta.name}
              />
              <h6>From contract: {props?.contract}</h6>
              <h6>Description: {props?.meta.description}</h6>
              <h6>Minted at: {props?.mintedAt}</h6>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default NftDetailsModal;
