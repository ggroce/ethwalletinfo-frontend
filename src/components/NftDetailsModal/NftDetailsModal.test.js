import { render } from '@testing-library/react';
import NftDetailsModal from './NftDetailsModal';

test('<NftDetailsModal />: takes a snapshot', () => {
  const { asFragment } = render(<NftDetailsModal />);
  expect(asFragment(<NftDetailsModal />)).toMatchSnapshot();
});

const mockNft = {
  id: '1',
  meta: {
    name: 'NFT',
    image: {
      url: {
        PREVIEW: 'https://example.com/preview.png',
        ORIGINAL: 'https://example.com/original.png',
      },
    },
  },
};
