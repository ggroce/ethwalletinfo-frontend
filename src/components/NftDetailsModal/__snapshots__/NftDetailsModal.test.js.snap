// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NftDetailsModal />: takes a snapshot 1`] = `
<DocumentFragment>
  <div
    class="nftdetails__background--darken"
  >
    <div
      class="nftdetails__wrapper--center"
    >
      <div
        class="nftdetails__outerwrapper"
      >
        <div
          class="nftdetails__innerwrapper"
        >
          <h5 />
          <img
            class="nftdetails__img"
            loading="lazy"
            title="Picture of NFT"
          />
          <h6>
            From contract: 
          </h6>
          <h6>
            Description: 
          </h6>
          <h6>
            Minted at: 
          </h6>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
